### DISCUSSION TITLE
Google Cloud KMS SSH Host Key Software Backed

### DISCUSSION THREAD

**Discussion by milesstoetzner** [smallstep/certificates#460] [Q&A] [answered]
Hello, 

the protection level of the ssh-host-key is hardcoded as `software` in `step-cloudkms-init`. See https://github.com/smallstep/certificates/blob/6c0cf99b2445228633377230871ae5f14f0e89ff/cmd/step-cloudkms-init/main.go#L237 

What is the reason for that?

Greetings,
Miles

**Comment by maraino** [comment, marked as answer]
It looks like we missed the flag variable when we added ssh support. We'll fix it, but this tool is basically for experimenting, you shouldn't relay on it directly for a real PKI. For example the subject of X.509 certificates are also hardcoded to Smallstep Root.

**Reply by maraino** [reply]
Fixed in a branch for an upcoming release 51ac286

**Reply by milesstoetzner** [reply]
yeah we noticed that too and assumed that this is because of the experimental status

**Reply by maraino** [reply]
Feel free to adapting this "script" for your needs, to create your PKI. It's probably easier than doing it using gcloud :)
